refactor(tests): replace string JSONs with `Json::Value`

mvadari opened the pull request:
## High Level Overview of Change

There are some tests that write out JSONs as a string instead of using the `Json::Value` library. This PR cleans that up. 

### Context of Change

Making the tests easier to read and edit

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)

### API Impact

N/A

## Test Plan

CI passes

pratikmankawde commented on src/test/app/TxQ_test.cpp:
```diff
@@ -3089,24 +3089,32 @@ class TxQPosNegFlows_test : public beast::unit_test::suite
             // account_info without the "queue" argument.
-            auto const info = env.rpc("json", "account_info", withoutQueue);
+            auto const info =
+                env.rpc("json", "account_info", to_string(withoutQueue));
```
Suggestion: Can we predefine these strings `json`, `account_info` at the beginning of this file and then reuse them?

mvadari replied:
Why? IMO that'd make the tests harder to read

pratikmankawde replied:
Defining it once will remove the repeated allocation of these items.

Why would it make the test harder to read? The variables will have the same name as the value (following the naming convention for such variables of-course). Something like `s_JSON`, `s_ACCOUNT_INFO`.

mvadari replied:
I think that would be more confusing, as you then have to go back and look at what those variables are defined as instead of just seeing it in the test. Also, this is the pattern used in all tests in this repo.

pratikmankawde replied:
We may need to revisit such patterns at some point. Such repeated allocations are suboptimal and makes it even harder to track and update them when needed. With such patterns, in an open source software, we are allowing external contributors to overlook such things and make things worse for the future contributors.

I believe following the naming conventions should solve the readability aspect.

mvadari replied:
Regardless, such a change should be discussed separately and made in a separate PR.

pratikmankawde commented on src/test/rpc/AccountLines_test.cpp:
```diff
@@ -541,13 +510,12 @@ class AccountLines_test : public beast::unit_test::suite
+        Json::Value linesEndParams;
+        linesEndParams[jss::account] = alice.human();
+        linesEndParams[jss::marker] =
+            linesBeg[jss::result][jss::marker].asString();
```
Question: Do you think calling `asString()` here might be redundant?
`linesEndParams[jss::marker] = linesBeg[jss::result][jss::marker]` might be enough?

mvadari replied:
Good catch - fixed

pratikmankawde commented on src/test/rpc/AccountLines_test.cpp:
```diff
@@ -751,12 +718,11 @@ class AccountLines_test : public beast::unit_test::suite
             // Get becky's objects just to confirm that they're symmetrical
+            Json::Value beckyObjectsParams;
+            beckyObjectsParams[jss::account] = becky.human();
+            beckyObjectsParams[jss::limit] = 200;
```
Suggestion: This could be out of scope for this PR, but what do these numbers mean? Why 200? If it's trivial, maybe it would be a good idea to add a-few comments explaining the test params?

mvadari replied:
I would say that's out of scope of this PR - my goal in this PR is to just convert the old `R"` strings to use `Json::Value`, I didn't do any analysis of what these tests do.

pratikmankawde commented on src/test/rpc/AccountLines_test.cpp:
```diff
@@ -1458,19 +1369,17 @@ class AccountLines_test : public beast::unit_test::suite
+        Json::Value linesEndParams;
+        linesEndParams[jss::account] = alice.human();
+        linesEndParams[jss::marker] =
+            linesBeg[jss::result][jss::marker].asString();
```
What does this comment `So the old marker should now be invalid.` mean? What will the `asString` here `linesBeg[jss::result][jss::marker].asString()` return? Do we need to call it explicitely? Or the assignment is sufficient?

mvadari replied:
I cleared out all the `asString`s, though those were there previously

pratikmankawde reviewed (commented):
Left some comments. Looks good otherwise.
I have very limited view of these changes. Not sure if this is the only place these are required or there are other places and they are to be done later?

pratikmankawde commented on src/test/rpc/AccountLines_test.cpp:
```diff
@@ -1267,7 +1267,7 @@ class AccountLines_test : public beast::unit_test::suite
 
             // Continue from the returned marker to make sure that works.
             BEAST_EXPECT(linesA[jss::result].isMember(jss::marker));
-            auto const marker = linesA[jss::result][jss::marker];
```
Why did you put back the `asString`?
Do we need to put it back on line no. 515 and 1133 as well?

mvadari replied:
I had to add them back because the build was failing and those `asString`s didn't need to be removed as a part of this PR.

pratikmankawde reviewed (approved):
Approving.

bthomee committed:
refactor: replace string JSONs with Json::Value

There are some tests that write out JSONs as a string instead of using the Json::Value library, which are cleaned up by this change.
